### ISSUE TITLE
Bug in close logic?

### ISSUE THREAD

**Issue by StephenAtty** [fmancardi/mantis-triggerclose#8]
The current logic to identify what to close does this

```
SELECT
				bht.bug_id,
				bt.summary
			FROM
				".db_get_table('mantis_bug_table')." AS bt
			INNER JOIN
				".db_get_table('mantis_bug_history_table')." AS bht
				ON
					bht.bug_id = bt.id
				AND
					bht.date_modified < %d
			INNER JOIN
				".db_get_table('mantis_user_table')." AS ut
				ON
					bt.handler_id = ut.id
				AND
					ut.access_level IN (%s)
			WHERE
				bt.status IN (%s)
				AND
				bt.category_id IN (%s)
			GROUP BY
				bht.bug_id",

```
But this will find any bug that is in a closable state that has an entry in the history table that pre-dates the cutoff date. 

We only want to find bugs where the last history timestamp is before the cut off date because if we take a bug from out of closed into a closable state (such as feedback) we create a new history record but the code as is will close the bug the next time it runs because it's in a closable state and has an history timestamp that pre-dates the cut off date

So I think this code needs to be

```
				$sql = sprintf("
			SELECT
				bt.id as bug_id,
				bt.summary
			FROM
				".db_get_table('mantis_bug_table')." AS bt
			INNER JOIN
				".db_get_table('mantis_user_table')." AS ut
				ON
					bt.handler_id = ut.id
				AND
					ut.access_level IN (%s)
			WHERE
				bt.status IN (%s)
				AND
				bt.category_id IN (%s)
				and %d >  (select max(bht.date_modified) from mantis_bug_history_table bht
					where
					bht.bug_id = bt.id
					)
			GROUP BY
				bh.id",
```

So basically only find records where the last date modified is before the cut off date.
